crates/sargon/src/system/sargon_os/transactions/sargon_os_transaction_analysis.rs

```diff
@@ -100,58 +112,68 @@ impl SargonOS {
             );
         }
 
-        Ok(TransactionToReview {
-            transaction_manifest,
-            execution_summary,
-        })
+        match subintent_manifest.as_enclosed() {
```

Comment:
I prefer:

```rust
let value =  subintent_manifest.as_enclosed().map(|manifest| PreAuthToReview::Enclosed(PreAuthEnclosedManifest { .. })).unwrap_or(PreAuthToReview::Open(PreAuthOpenManifest { ..}));
Ok(value)
```
```

Reply:
There will be additional setup for the case of enclosed before creating the actual result, as exemplified in the comment

Reply:
My point was I prefer not having multiple Ok and trying to use functional methods such as map and unwrap if possible :)

Reply:
not quite possible in this case, we will have an async function that would resolve the enclosed case. Also something like `unwrap_or` in this chain, feels like the Open pre-auth is simply a fallback to the Enclosed pre-auth, at least for me :).
Agree for the Ok, I would then use this:
```rust
let pre_auth_to_review = match subintent_manifest.as_enclosed() {
      Some(manifest) => {
          // TODO: perform the actual analysis
          // let gateway_client = GatewayClient::new(
          //     self.clients.http_client.driver.clone(),
          //     network_id,
          // );
          // let response = self.get_subintent_preview(manifest, gateway_client, nonce).await?;

          PreAuthToReview::Enclosed(PreAuthEnclosedManifest {
              manifest: subintent_manifest,
              summary: ExecutionSummary::sample_stokenet(),
          })
      }
      None => PreAuthToReview::Open(PreAuthOpenManifest {
          manifest: subintent_manifest,
          summary: summary,
      }),
  };

  Ok(pre_auth_to_review)
```